azurerm/data_source_kubernetes_cluster.go

```diff
@@ -459,9 +477,21 @@ func dataSourceArmKubernetesClusterRead(d *schema.ResourceData, meta interface{}
 	if props := resp.ManagedClusterProperties; props != nil {
 		d.Set("dns_prefix", props.DNSPrefix)
 		d.Set("fqdn", props.Fqdn)
+		d.Set("private_fqdn", props.PrivateFQDN)
 		d.Set("kubernetes_version", props.KubernetesVersion)
 		d.Set("node_resource_group", props.NodeResourceGroup)
 
+		if props.APIServerAccessProfile != nil {
+			apiServerAuthorizedIPRanges := utils.FlattenStringSlice(props.APIServerAccessProfile.AuthorizedIPRanges)
+			if err := d.Set("api_server_authorized_ip_ranges", apiServerAuthorizedIPRanges); err != nil {
+				return fmt.Errorf("Error setting `api_server_authorized_ip_ranges`: %+v", err)
+			}
+
+			if err := d.Set("private_link_enabled", props.APIServerAccessProfile.EnablePrivateCluster); err != nil {
+				return fmt.Errorf("Error setting `private_link_enabled`: %+v", err)
+			}
```

Comment:
we can remove this error check here - to make this:

```suggestion
			d.Set("private_link_enabled", props.APIServerAccessProfile.EnablePrivateCluster)
```

Reply:
Hi Tom

There's been a change in the API, now `private_link_enabled` (EnablePrivateCluster)  is a field of   `props.APIServerAccessProfile` so if we are adding a nil check as you sugested for `APIServerAccessProfile` ,the check for `EnablePrivateCluster`  it should be inside. Othewise we could get the same nil.

Reply:
let me know what you think.

Reply:
OK, clarified.